### PR TITLE
feat:Failed to use the absolute path to start kuiper #229

### PR DESCRIPTION
**Pull request by EMQmyd**

### CODE REVIEW

File: common/util.go [hunk 1]
```diff
@@ -159,8 +163,33 @@ func GetDataLoc() (string, error) {
 	return GetLoc(data_dir)
 }
 
+func getAbsPath(subdir string) (dir string, err error) {
+	switch true {
```
Comment (ngjaying) [inline common/util.go:167]: Seems like changing the previous semantic with contains.  "aetc" will now also return etc path, I don't think it is expected.

File: common/util.go [hunk 2]
```diff
@@ -18,13 +18,17 @@ import (
...
-	data_dir      = "/data/"
-	log_dir       = "/log/"
-	StreamConf    = "kuiper.yaml"
-	KuiperBaseKey = "KuiperBaseKey"
-	MetaKey       = "__meta"
+	logFileName    = "stream.log"
+	etc_dir        = "/etc/"
+	data_dir       = "/data/"
+	log_dir        = "/log/"
+	abs_etc_dir    = "/etc/kuiper/"
```
Comment (ngjaying) [inline common/util.go:25]: How about using map for these constant abs path

File: common/util.go [hunk 3]
```diff
@@ -159,8 +163,33 @@ func GetDataLoc() (string, error) {
...
+	}
+	if _, err = os.Stat(dir); os.IsExist(err) {
+		return dir, nil
+	}
+	return "", err
+}
+
 func GetLoc(subdir string) (string, error) {
-	dir, err := os.Getwd()
+	dir, err := getAbsPath(subdir)
```
Comment (ngjaying) [inline common/util.go:188]: Need to check with @zhanghongtong which is the highest priority: absolute or relative path

**Comment by jinfahua** [comment]
2 tests are failed @EMQmyd , it seems related to your fix.

**Pull request merged**